aea/cli/run.py

```diff
@@ -18,11 +18,12 @@
 #
 # ------------------------------------------------------------------------------
 """Implementation of the 'aea run' subcommand."""
-from contextlib import contextmanager
+from contextlib import contextmanager, nullcontext
 from pathlib import Path
 from typing import Generator, List, Optional, Sequence, Tuple, cast
 
 import click
+import memray  # type: ignore
```

Comment:
This is a problem. Means all aea's need memray even if they don't use it - not the way to do it (and even if we wanted to do it, you'd need to mention it in setup.py as a dependency).

I also don't think we should mix the two profilers as the one affects the other as well as the aea.

The one way we could incorporate it, if at all, is to use a flag and then have lazy loaded memray; with an exception handling of not installed case and nice error message.

Reply:
A neater way still, is to use the setup.py extras mechanism; but might be overkill here.

Reply:
Ok, I've updated the PR now. I had abandoned it as I understood we were not interested in it.